src/profile/components/multiItemRow/MultiItemRow.tsx

```diff
@@ -0,0 +1,122 @@
+import { TextInput } from 'hds-react';
+import React from 'react';
+import { Field, Formik, FormikProps, Form } from 'formik';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+
+import styles from './multiItemRow.module.css';
+import { EditDataValue, EmailValue, PhoneValue } from '../../helpers/editData';
+import { createFormFieldHelpers } from '../../helpers/formik';
+import { phoneSchema, emailSchema } from '../../../common/schemas/schemas';
+import FormButtons from '../formButtons/FormButtons';
+import EditButtons from '../editButtons/EditButtons';
+import commonFormStyles from '../../../common/cssHelpers/form.module.css';
+import SaveIndicator from '../saveIndicator/SaveIndicator';
+import { useCommonEditHandling } from '../../hooks/useCommonEditHandling';
+import { RowItemProps } from '../multiItemEditor/MultiItemEditor';
+import { getFormFields } from '../../helpers/formProperties';
+
+type EmailAndPhoneFormikValue = { value: string };
+
+function MultiItemRow(props: RowItemProps): React.ReactElement {
+  const { data, testId, dataType } = props;
+  const { t } = useTranslation();
+  const {
+    isEditing,
+    currentAction,
+    actionHandler,
+    isNew,
+  } = useCommonEditHandling(props);
+  const { value, primary } = data;
+  const schema = dataType === 'phones' ? phoneSchema : emailSchema;
+  const inputValue: string =
+    (dataType === 'phones'
+      ? (value as PhoneValue).phone
+      : (value as EmailValue).email) || '';
+
+  const inputId = `${testId}-value`;
+  const formFields = getFormFields(dataType);
+  const { hasFieldError, getFieldErrorMessage } = createFormFieldHelpers<
+    EmailAndPhoneFormikValue
+  >(t, isNew);
+  const convertFormPropsToEditDataValue = (
+    formValues: EmailAndPhoneFormikValue
+  ): Partial<EditDataValue> => {
+    const formValue = formValues.value;
+    const propName = dataType === 'phones' ? 'phone' : 'email';
+    return {
+      [propName]: formValue,
+    };
+  };
+  if (isEditing) {
+    return (
+      <div
+        className={classNames([
+          commonFormStyles.contentWrapper,
+          styles.rowContentWrapper,
+        ])}
+      >
+        <Formik
+          initialValues={{
+            value: inputValue,
+          }}
+          onSubmit={async values => {
+            const editDataValues = convertFormPropsToEditDataValue(values);
+            await actionHandler('save', editDataValues);
+          }}
+          validationSchema={schema}
+        >
+          {(formikProps: FormikProps<EmailAndPhoneFormikValue>) => (
+            <Form>
+              <div className={styles.editableRow}>
+                <Field
+                  name="value"
+                  id={inputId}
+                  maxLength={formFields.value.max as number}
+                  as={TextInput}
```

Comment:
Should we apply the HDS PhoneInput component here when type is phone? https://hds.hel.fi/components/phone-input

One benefit is that it adds the type=tel to the input, which then helps mobile users by opening the number keyboard

Reply:
NumberInput, PhoneInput and other HDS fixes mentioned here were released after this was originally created. HDS update would probably cause a lot of changes, so did not want to update it in this PR and make it larger. I'd rather make a new ticket for HDS updates.